src/gui/inputdevices/InputContext.cpp

```diff
@@ -70,70 +70,67 @@ auto InputContext::eventCallback(GtkWidget* widget, GdkEvent* event, InputContex
         this->getSettings()->transactionStart();
-        auto deviceClassOption = this->getSettings()->getDeviceClassForDevice(string(event->deviceName), inputSource);
+        auto deviceClassOption = this->getSettings()->getDeviceClassForDevice(string(event.deviceName), inputSource);
         this->getSettings()->setDeviceClassForDevice(sourceDevice, deviceClassOption);
         this->getSettings()->transactionEnd();
     }
 
     // We do not handle scroll events manually but let GTK do it for us
-    if (event->type == SCROLL_EVENT) {
+    if (event.type == SCROLL_EVENT) {
         // Hand over to standard GTK Scroll / Zoom handling
         return false;
     }
 
     // Deactivate touchscreen when a pen event occurs
-    this->getView()->getHandRecognition()->event(event->deviceClass);
+    this->getView()->getHandRecognition()->event(event.deviceClass);
 
     // Get the state of all modifiers
-    this->modifierState = event->state;
+    this->modifierState = event.state;
 
     // separate events to appropriate handlers
     // handle tablet stylus
-    if (event->deviceClass == INPUT_DEVICE_PEN || event->deviceClass == INPUT_DEVICE_ERASER) {
-        return this->stylusHandler->handle(event);
+    if (event.deviceClass == INPUT_DEVICE_PEN || event.deviceClass == INPUT_DEVICE_ERASER) {
+        return this->stylusHandler->handle(&event);
     }
 
-    if (event->deviceClass == INPUT_DEVICE_MOUSE_KEYBOARD_COMBO) {
-        return this->mouseHandler->handle(event) || this->keyboardHandler->handle(event);
+    if (event.deviceClass == INPUT_DEVICE_MOUSE_KEYBOARD_COMBO) {
+        return this->mouseHandler->handle(&event) || this->keyboardHandler->handle(&event);
     }
 
     // handle mouse devices
-    if (event->deviceClass == INPUT_DEVICE_MOUSE) {
-        return this->mouseHandler->handle(event);
+    if (event.deviceClass == INPUT_DEVICE_MOUSE) {
+        return this->mouseHandler->handle(&event);
     }
 
     // handle touchscreens
-    if (event->deviceClass == INPUT_DEVICE_TOUCHSCREEN) {
+    if (event.deviceClass == INPUT_DEVICE_TOUCHSCREEN) {
         // trigger touch drawing depending on the setting
         if (this->touchWorkaroundEnabled) {
-            return this->touchDrawingHandler->handle(event);
+            return this->touchDrawingHandler->handle(&event);
         }
 
-        return this->touchHandler->handle(event);
+        return this->touchHandler->handle(&event);
     }
 
     // handle keyboard
-    if (event->deviceClass == INPUT_DEVICE_KEYBOARD) {
-        return this->keyboardHandler->handle(event);
+    if (event.deviceClass == INPUT_DEVICE_KEYBOARD) {
+        return this->keyboardHandler->handle(&event);
     }
 
-    if (event->deviceClass == INPUT_DEVICE_IGNORE) {
+    if (event.deviceClass == INPUT_DEVICE_IGNORE) {
         return true;
     }
-
-    delete event;
-
-    // We received an event we do not have a handler for
+    g_warning("We received an event we do not have a handler for");
```

Comment:
Can't this get very spammy? Maybe we should hide this behind the `DEBUG_INPUT` flag.

Reply:
Mhh yes indeed, I wanted users to report that, since we should know when we miss events unintentionaly.